c2corg_api/views/validation.py

```diff
@@ -604,6 +604,25 @@ def _get_linked_document_ids(associations):
         ])
 
 
+def _are_valid_waypoints(associations_in, main_document_type, document_key):
+    """ Check if waypoints can be associated to the main document
+    """
+    if main_document_type != ROUTE_TYPE or document_key != 'waypoints':
+        return True
+
+    waypoints_id = [doc['document_id'] for doc in associations_in[document_key]]
+
+    # load the waypoint type for each linked waypoint
+    if waypoints_id:
+        query_waypoints_with_type = DBSession. \
+            query(Waypoint.waypoint_type). \
+            filter(Waypoint.document_id.in_(waypoints_id))
+        for waypoint in query_waypoints_with_type.all():
+            if waypoint.waypoint_type == 'climbing_indoor':
+                return False
```

Comment:
maybe the function could return the error description instead of a boolean, so that it can be more easily handled on line 648 if several cases arise?

currently, the error handling there is not really clear because the function name doesn't explicit it handles a single case

Reply:
I've changed the fonction name, so it is more clear that it handles a single case